sdk/spring-sdk-testkit/src/main/java/kalix/springsdk/testkit/KalixConfigurationTest.java

```diff
@@ -52,6 +53,10 @@ public WebClient createWebClient(KalixTestKit kalixTestKit) {
     return WebClient.builder()
         .baseUrl("http://localhost:" + kalixTestKit.getPort())
         .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+        .codecs(configurer -> {
+          configurer.defaultCodecs().jackson2JsonEncoder(
+            new Jackson2JsonEncoder(JsonSupport.getObjectMapper(), MediaType.APPLICATION_JSON));
```

Comment:
I just realised that the Jackson Protobuf lib has a limitation. 

A type with two complex fields that have the same simple class name will generate an error. For example:

```java

interface A {
  record Foo(int i)
}

interface B {
  record Foo(String s)
}

class Bar {
  public A.Foo fooA;
  public B.Foo fooB;
}
```

This will generate a message for `Bar` and two messages for `Foo`. 
The `Foo` messages will identical. 

They will have either a `int32` field or a `string` field. Basically, it will consider that they are the same.

That's because protobuf messages doesn't have namespace.

We can't do much about it. We can validate and throw an exception instructing users how to fix though.

Reply:
I created an issue to track this. https://github.com/lightbend/kalix-jvm-sdk/issues/1432

Reply:
2 nice finds! 👍